qa/rpc-tests/mempool_push.py

```diff
@@ -0,0 +1,163 @@
+
+    def setup_network(self, split=False):
+        mempoolConf = [
+            ["-blockprioritysize=2000000"],
+            ["-blockprioritysize=2000000", "-maxmempool=8080", "-limitancestorsize=%d" % (BCH_UNCONF_SIZE_KB*2), "-limitdescendantsize=%d" % (BCH_UNCONF_SIZE_KB*2), "-limitancestorcount=%d" % (BCH_UNCONF_DEPTH*2), "-limitdescendantcount=%d" % (BCH_UNCONF_DEPTH*2), "-net.unconfChainResendAction=2"],
+            ["-blockprioritysize=2000000", "-limitdescendantcount=1000", "-limitancestorcount=1000","-limitancestorsize=1000", "-limitdescendantsize=1000", "-net.unconfChainResendAction=2"],
+            ["-blockprioritysize=2000000", "-limitancestorsize=150","-net.unconfChainResendAction=2"]
+            ]
+        self.nodes = start_nodes(4, self.options.tmpdir, mempoolConf)
+        connect_nodes_full(self.nodes)
+        self.is_network_split=False
+        self.sync_blocks()
+
+    def run_test (self):
+        # kick us out of IBD mode since the cached blocks will be old time so it'll look like our blockchain isn't up to date
+        # if we are in IBD mode, we don't request incoming tx.
+        self.nodes[0].generate(1)
+
+        logging.info("ancestor count test")
+        bal = self.nodes[1].getbalance()
+        addr = self.nodes[1].getnewaddress()
+
+        txhex = []
+        for i in range(0,BCH_UNCONF_DEPTH*2):
+          try:
+            txhex.append(self.nodes[1].sendtoaddress(addr, bal-1))  # enough so that it uses all UTXO, but has fee left over
+          except JSONRPCException as e: # an exception you don't catch is a testing error
+              print(str(e))
+              raise
+        assert_equal(self.nodes[0].getmempoolinfo()["size"], BCH_UNCONF_DEPTH)
+        assert_equal(self.nodes[1].getmempoolinfo()["size"], BCH_UNCONF_DEPTH*2)
+        # Set small to commit just a few tx so we can see if the missing ones get pushed
+        self.nodes[0].set("mining.blockSize=6000")
+        blk = self.nodes[0].generate(1)[0]
+        blkhex = self.nodes[0].getblock(blk)
+        waitFor(DELAY_TIME, lambda: self.nodes[0].getmempoolinfo()["size"] == BCH_UNCONF_DEPTH)
+        # generate the block somewhere else and see if the tx get pushed
+        self.nodes[2].set("mining.blockSize=4000")
+        blk2 = self.nodes[2].generate(1)[0]
+        waitFor(DELAY_TIME, lambda: self.nodes[0].getmempoolinfo()["size"] == BCH_UNCONF_DEPTH)
+
+        waitFor(DELAY_TIME, lambda: self.nodes[1].getbestblockhash() == blk2)  # make sure its settled so we can get a good leftover count for the next test.
+        unconfLeftOver = self.nodes[1].getmempoolinfo()["size"]
+        assert(unconfLeftOver >= BCH_UNCONF_DEPTH)  # if someone bumps the BCH network unconfirmed depth, you need to build a bigger unconf chain
+        # Let's consume all BCH_UNCONF_DEPTH tx
+        self.nodes[0].set("mining.blockSize=8000000")
+
+        waitFor(DELAY_TIME, lambda: len(self.nodes[0].getblocktemplate()["transactions"])>=BCH_UNCONF_DEPTH)
```

Comment:
wouldn't the above be more "precise" if using `= BCH_UNCONF_DEPTH + 1` rather than `>= BCH_UNCONF_DEPTH` (used `+ 1` to take into account the coinbase tx)?

I mean node 0 has a max len for unconf txs chain of `BCH_UNCONF_DEPTH`, we only generated a chain of `BCH_UNCONF_DEPTH * 2` txs in the beginning of the test. So it seems impossible to me for node0 to produce a block template with anything different than `BCH_UNCONF_DEPTH + 1` (taking into account that max gen block size is set to 8MB)